doofinder/includes/multilanguage/class-language-plugin.php

```diff
@@ -2,72 +2,103 @@
+     *
+     * @return string Lang code of current selected language.
+     */
+    abstract public function get_current_language();
 
-	/**
-	 * Get all posts ids of a given language.
-	 *
-	 * @param string $language_code
-	 * @param string $post_type
-	 *
-	 * @return int[] List of ids.
-	 */
-	abstract public function get_posts_ids( $language_code, $post_type);
+    /**
+     * Retrieve the base language of the site.
+     *
+     * This is important because the behavior of the site (e.g. language-specific
+     * option names) should be the same as if there was no multilanguage plugin
+     * installed.
+     *
+     * @return string Lang code of the base (primary) language of the site.
+     */
+    abstract public function get_default_language();
 
-	/**
-	 * Retrieve the name of the wordpress option
-	 * for the current languages.
-	 *
-	 * Some fields in Doofinder settings will have different values,
-	 * depending on language.
-	 *
-	 * @param string $base
-	 *
-	 * @return string
-	 */
-	public function get_option_name( $base ) {
-		$language_code = $this->get_active_language();
-		if ( ! $language_code ) {
-			return $base;
-		}
+    /**
+     * Retrieve the base language of the site.
+     *
+     * This is important because the behavior of the site (e.g. language-specific
+     * option names) should be the same as if there was no multilanguage plugin
+     * installed.
+     *
+     * @return string Lang code of the base (primary) language of the site.
+     */
+    abstract public function get_base_language();
 
-		$base_language = $this->get_base_language();
-		if ( $language_code === $base_language ) {
-			return $base;
-		}
+    /**
+     * Retrieve the home URL for the given language.
+     *
+     * @since 1.0.0
+     * @param string $language Language to retrieve home URL for.
+     * @return string Home URL.
+     */
+    abstract public function get_home_url($language);
 
-		return "{$base}_{$language_code}";
-	}
-}
\ No newline at end of file
+    /**
+     * Retrieve the name of the wordpress option
+     * for the current languages.
+     *
+     * Some fields in Doofinder settings will have different values,
+     * depending on language.
+     *
+     * @param string $base
+     *
+     * @return string
+     */
+    // public function get_option_name( $base ) {
+    // 	$language_code = $this->get_active_language();
+    // 	if ( ! $language_code ) {
+    // 		return $base;
+    // 	}
+
+    // 	$base_language = $this->get_base_language();
+    // 	if ( $language_code === $base_language ) {
+    // 		return $base;
+    // 	}
+
+    // 	return "{$base}_{$language_code}";
+    // }
+
+    /**
+     * Retrieve the name of the wordpress option
+     * for the current languages.
+     *
+     * Some fields in Doofinder settings will have different values,
+     * depending on language.
+     *
+     * @param string $base
+     *
+     * @return string
+     */
+    //abstract public function get_option_name( $base );
```

Comment:
All this commented code shouldn't be deleted? Also, this class seems more like an interface to me

Reply:
I've removed the commented code.